### PR TITLE
Fix nodelist replace bug

### LINKED ISSUE #1290: .replace() method sometimes doesn't properly replace AST nodes
Take following call statement: `callSomeFun(r -> r instanceof SomeType);`.

Let's call the lambda expression argLambdaExpr and create a new MethodCallExpr to lambdaWrapper() called methodCallLambdaReplacement. When I call `argLambdaExpr.replace(methodCallLambdaReplacement)` the resulting callSomeFun-expression should be:

`callSomeFun(lambdaWrapper());` 

In fact, in the `arguments` list the lambda expression in the MethodCallExpr is replaced with our new methodCallLambdaReplacement expression. However, the `childNodes` element of the MethodCallExpr still contains the lambda expression (in addition to the new method call expression). I believe the lambda expression is supposed to be be removed from the childNodes list by the .replace() method as well.

### PR DESCRIPTION
**Pull request by matozoid**
Fixes #1290

### CODE REVIEW

File: javaparser-testing/src/test/java/com/github/javaparser/ast/expr/MethodCallExprTest.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    public void replaceLambdaIssue1290() {
+        MethodCallExpr methodCallExpr = parseExpression("callSomeFun(r -> r instanceof SomeType)").asMethodCallExpr();
+        LambdaExpr lambdaExpr = methodCallExpr.getArgument(0).asLambdaExpr();
+        MethodCallExpr lambdaWrapper = new MethodCallExpr("lambdaWrapper");
+        lambdaExpr.replace(lambdaWrapper);
+        
+        assertEquals(2, methodCallExpr.getChildNodes().size());
+    }
+
+}
```
Comment (ftomassetti) [inline javaparser-testing/src/test/java/com/github/javaparser/ast/expr/MethodCallExprTest.java]: newline :D

File: javaparser-testing/src/test/java/com/github/javaparser/ast/expr/MethodCallExprTest.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+public class MethodCallExprTest {
+    
+    @Test
+    public void replaceLambdaIssue1290() {
+        MethodCallExpr methodCallExpr = parseExpression("callSomeFun(r -> r instanceof SomeType)").asMethodCallExpr();
+        LambdaExpr lambdaExpr = methodCallExpr.getArgument(0).asLambdaExpr();
+        MethodCallExpr lambdaWrapper = new MethodCallExpr("lambdaWrapper");
+        lambdaExpr.replace(lambdaWrapper);
+        
+        assertEquals(2, methodCallExpr.getChildNodes().size());
```
Comment (ftomassetti) [inline javaparser-testing/src/test/java/com/github/javaparser/ast/expr/MethodCallExprTest.java:18]: Could we add a check on the node being replaced having a null parent?

**Review by ftomassetti — APPROVED** [review]
Looks good!

File: javaparser-testing/src/test/java/com/github/javaparser/ast/expr/MethodCallExprTest.java (hunk 1 above)
Comment (matozoid) [reply javaparser-testing/src/test/java/com/github/javaparser/ast/expr/MethodCallExprTest.java]: No :-P

**Pull request merged**